Rubberduck.RibbonDispatcher/RibbonFactory.cs

```diff
@@ -0,0 +1,153 @@
+        /// <summary>TODO</summary>
+        internal void ActivateTab(string ControlId)           => _ribbonUI.ActivateTab(ControlId);
+
+        /// <summary>Returns a new Ribbon Group ViewModel instance.</summary>
+        [SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed", Justification="Matches COM usage.")]
+        public RibbonGroup NewRibbonGroup(string ItemId, bool Visible = true, bool Enabled = true)
+            => Add(new RibbonGroup(ItemId, _resourceManager, Visible, Enabled));
+
+        /// <summary>Returns a new Ribbon ActionButton ViewModel instance.</summary>
+        [SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed", Justification="Matches COM usage.")]
+        public RibbonButton NewRibbonButton(string ItemId, bool Visible = true, bool Enabled = true,
+            RdControlSize Size      = rdLarge,
+            IPictureDisp  Image     = null,
+            bool          ShowImage = true,
+            bool          ShowLabel = true
+        ) => Add(new RibbonButton(ItemId, _resourceManager, Visible, Enabled, Size, new ImageObject(Image), ShowImage, ShowLabel));
+        /// <summary>Returns a new Ribbon ActionButton ViewModel instance.</summary>
+        [SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed", Justification="Matches COM usage.")]
+        public RibbonButton NewRibbonButtonMso(string ItemId, bool Visible = true, bool Enabled = true,
+            RdControlSize Size      = rdLarge,
+            string        ImageMso  = "Unknown",
+            bool          ShowImage = true,
+            bool          ShowLabel = true
+        ) => Add(new RibbonButton(ItemId, _resourceManager, Visible, Enabled, Size, new ImageObject(ImageMso), ShowImage, ShowLabel));
+
+        /// <summary>Returns a new Ribbon ToggleButton ViewModel instance.</summary>
+        [SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed", Justification="Matches COM usage.")]
+        public RibbonToggleButton NewRibbonToggle(string ItemId, bool Visible = true, bool Enabled = true,
+            RdControlSize Size      = rdLarge,
+            IPictureDisp  Image     = null,
+            bool          ShowImage = true,
+            bool          ShowLabel = true
+        ) => Add(new RibbonToggleButton(ItemId, _resourceManager, Visible, Enabled, Size, new ImageObject(Image), ShowImage, ShowLabel));
+        /// <summary>Returns a new Ribbon ToggleButton ViewModel instance.</summary>
+        [SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed", Justification="Matches COM usage.")]
+        public RibbonToggleButton NewRibbonToggleMso(string ItemId, bool Visible = true, bool Enabled = true,
+            RdControlSize Size      = rdLarge,
+            string        ImageMso  = "Unknown",
+            bool          ShowImage = true,
+            bool          ShowLabel = true
```

Comment:
Distractingly inconsistent, please fix.

Reply:
I should sic Joe Celko on you - he is truly fanatical about the readability improvement gained from use of vertical ribbons. In this case, they are easily eliminated in a module by the triple key chord Ctrl-A, Ctrl-C, Ctrl-V.

Reply:
Joe Celko can stick to SQL, I'll take Robert C. Martin's advice over any other:

> Eliminate unnecessary, unhelpful alignment that draw your attention to the wrong thing (in this case, variable names rather than type)
(followed by a block of declarations formatted the way you have them)